frontend/src/views/admin/matching/applicant-view/grid/grid-item.tsx

```diff
@@ -0,0 +1,396 @@
+                </button>
+                {applicantMatch.status &&
+                    ["hidden", "applied", "starred"].includes(
+                        applicantMatch?.status
+                    ) && (
+                        <button
+                            className="dropdown-item"
+                            onClick={() => {
+                                updateApplicantMatch(
+                                    "staged-assigned",
+                                    position.hours_per_assignment || 0
+                                );
+                                setShow(false);
+                            }}
+                        >
+                            Assign to <b>{position.position_code}</b> (
+                            {position.hours_per_assignment || 0})
+                        </button>
+                    )}
+                {applicantMatch.status === "staged-assigned" && (
+                    <button
+                        className="dropdown-item"
+                        onClick={() => {
+                            setShowChangeHours(true);
+                            setShow(false);
+                        }}
+                    >
+                        Change assigned hours
+                    </button>
+                )}
+                {applicantMatch.status === "staged-assigned" && (
+                    <button
+                        className="dropdown-item"
+                        onClick={() => {
+                            updateApplicantMatch("applied");
+                            setShow(false);
+                        }}
+                    >
+                        Unassign from <b>{position.position_code}</b>
+                    </button>
+                )}
+                {applicantMatch.status !== "assigned" &&
+                    applicantMatch.status !== "hidden" && (
+                        <button
+                            className="dropdown-item"
+                            onClick={() => {
+                                updateApplicantMatch("hidden");
+                                setShow(false);
+                            }}
+                        >
+                            Hide from <b>{position.position_code}</b>
+                        </button>
+                    )}
+                {applicantMatch.status &&
+                    ["hidden", "applied", "starred"].includes(
+                        applicantMatch?.status
+                    ) && (
+                        <button
+                            className="dropdown-item"
+                            onClick={() => {
+                                hideApplicantFromAll();
+                                setShow(false);
+                            }}
+                        >
+                            Hide from all courses
+                        </button>
+                    )}
+                {applicantMatch.status === "hidden" && (
```

Comment:
My original thought was to have it in case a user accidentally clicks on hide. I also think I'm missing an "unhide from all courses" button as a counterpart to "hide from all courses", but let me know your thoughts on both.